storage/libradb/src/lib.rs

```diff
@@ -673,6 +673,17 @@ impl DbReader for LibraDB {
 
         Ok(tree_state)
     }
+
+    fn get_block_timestamp(&self, version: u64) -> Result<u64> {
+        // rely on the fact that blockmetadata txn always precedes user txn.
+        for i in (1..=version).rev() {
+            if let Transaction::BlockMetadata(t) = self.transaction_store.get_transaction(i)? {
+                return Ok(t.into_inner()?.1);
+            }
+        }
+        // genesis timestamp is 0
+        Ok(0)
+    }
```

Comment:
@christinacelee  what's the use case for this API? Will it be called much? Is linear search like this acceptable performance wise?

Reply:
(I can try to expose the .prev() interface on the iterator though, which will eliminate most of the random IOs here.)

Reply:
.prev() still makes it a linear search, just with a much smaller space to traverse. Do we not have an index of closest metadata block?

Reply:
I think the iterator loads in blocks, and .prev() won't issue another IO until the loaded block of bytes are exhausted. 

The index needed is described here https://github.com/libra/libra/issues/2338#issuecomment-606211862, we may implement it if use case requires it.

Reply:
this API will probably used most for querying for latest version to check how up-to-date your FN is on sync status, querying past versions will probably be called even less. But since this is a publicly open endpoint (not just something we're using internally), I don't think we should settle for this solution

Reply:
We agreed offline that linear search via a proper db iterator is probably good considering size of the RocksDB data block, the size of a avg transaction and number of transactions in a libra block. Let's land this and (I'll) upgrade it by exposing `.prev()` on the iter.

Reply:
sounds good!